Mixxx 2.4 press information / release blogpost (#325)
mixxxdj/website: content/news/XXXX-XX-XX-mixxx-2-4-0-features.md

This lets users prepare their tracks on a laptop using Mixxx and export them to a USB stick for use on a standalone controller at their gig.

### Saved Loops

Mixxx 2.4 adds support for Saved Loops.
If you already have tracks with Saved Loops from Serato DJ, Mixxx 2.4 will automatically read them from the track metadata.

@Video(https://www.youtube.com/watch?v=f8hI_uv9zHY)

### Effects

Mixxx 2.4 introduces Effect Chains, a set of effects which can be saved and reloaded with all their parameters and individual Meta Knob mappings.
Effect Chains can also be exported and imported, which allows you to share them with other DJs.
Once you've created an Effect Chain, you can define individual Super Knob mappings which allows you to use the chain as a Quick-Effect, where you take creative control with a single knob. Effect parameters can be reordered and or hidden.
In the effect units, Mixxx 2.4 also displays the value and unit of effect knob parameters while they are being adjusted.

### Track Menu

The track menu in Mixxx 2.4 is enhanced by several new features:

- Select loaded track in library
- Analyze/re-analyze with variable/constant BPM on a per-track basis

> - Looking up track metadata at Discogs, Soundcoud and LastFM

Reviewer: ```suggestion
- Looking up track metadata at Discogs, SoundCloud and LastFM
```

title: Mixxx 2.4 Released
authors: Mixxx Team
status: draft
tags: release, 2.4
comments: yes

We are proud to announce Mixxx 2.4, a new major version of our free open source DJ software, [available right now](https://mixxx.org/download/). It's been two years since the last release due to the large number of great new features and important changes under the hood.

The 2.4 release includes the following highlights:

> ### EngineOS export

Reviewer 2: ```suggestion
### EngineOS Export
```

### Saved Loops

Mixxx 2.4 adds support for Saved Loops, which can be assigned to any Hot Cue slot.
If you already have tracks from Serato DJ with Saved Loops, Mixxx 2.4 will import them automatically.

@Video(https://www.youtube.com/watch?v=f8hI_uv9zHY)

### Effect Chains

Mixxx 2.4 introduces Effect Chains, a set of effects which can be saved and reloaded with all their parameters and individual Meta Knob mappings.
Effect Chains can also be exported and imported, which allows you to share them with other Mixxx DJs.
Once you've created an Effect Chain, you can create a custom Super Knob mapping which turns the chain into a Quick-Effect, where you take creative control with a single knob.
Effect parameters can be reordered and or hidden.
In the effect units, Mixxx 2.4 also displays the value and unit of effect knob parameters while they are being adjusted.

### Track Menu Enhancements

The track menu in Mixxx 2.4 is enhanced by several new features:

- Select loaded track in library.
- Analyze/re-analyze with variable/constant BPM on a per-track basis.

> - Looking up track metadata at Discogs, Soundcoud and LastFM.

Reviewer: ```suggestion
- Look up track metadata at Discogs, SoundCloud and LastFM.
```

> - Select loaded track in library.
> - Analyze/re-analyze with variable/constant BPM on a per-track basis.
> - Looking up track metadata at Discogs, Soundcoud and LastFM.
> - Remove track files from disk.

Reviewer: ```suggestion
- Select loaded track in library
- Analyze/re-analyze with variable/constant BPM on a per-track basis
- Looking up track metadata at Discogs, SoundCloud and LastFM
- Remove track files from disk (does 'Move to Trash' with Qt >= 5.15)
```

Author: I'd prefer to keep the dots since those are sentences and not short clauses.  If it was a short list of nouns then removing the periods would make sense.

Reviewer: okay. see my new suggestion (typo + Looking -> Look)

### Sync Lock Improvements

Mixxx 2.4 adds support for manually selecting the Sync Leader by clicking on the new button with the crown symbol next to the Sync button.
This mode is useful when tracks have variable BPM.
All decks with sync enabled will follow the changing BPM of the Sync Leader.
@Video(https://www.youtube.com/watch?v=REiG8DBKr-M)

> ### Many other changes

Reviewer 2: ```suggestion
### Many Other Changes
```

##### Soundstretcher / Key-Shift Algorithms

Mixxx 2.4 now supports three different key shift algorithms:

- Soundtouch (Basic audio quality - medium CPU usage)
- Rubberband R2 (Good audio quality - higher CPU usage)
- Rubberband R3 (High audio quality - very high CPU usage)

> ##### Major rework of the Waveform Rendering code

Reviewer 2: This is too long for a headline. Maybe put it in the text body. 

```suggestion
##### Waveform Rendering
```

Author: I want to keep verbs in these bullets so it's clear what type of change occurred.

"Waveform Rendering Reworked" for instance

The waveform display has been rewritten, solving some longstanding performance issues, especially on macOS.

> ##### Native macOS ARM builds for Apple silicon

Reviewer 2: ```suggestion
##### Native macOS ARM builds
```

Reviewer 2: Is this a headline without a text paragraph or is github displaying it incorrectly?

Reviewer 3: Apple Silicon is the term normal users know, not ARM. And this page addresses users and media, not developers.
Maybe we should break it into two lines.

Reviewer 2: Like this? 

```suggestion
##### Apple Silicon Support

Mixxx now supports macOS ARM systems ("Apple silicon") such as M1 natively and provides installation packages for those architectures.
For macOS users, this increases the stability and performance of Mixxx on Apple silicon. 
```

Author: sounds good

##### Controller Backend

Modernized Javascript ES7 based controller mapping system.
HID backend rework.

## Mixxx 2.4 Factsheet

> ##### Supported platforms:

Reviewer 2: ```suggestion
##### Supported Platforms
```

Reviewer 2: I just noticed: the headline levels don't match up. The parent section uses `##` so this one should use `###` (this does not videolate the spec but is discouraged: https://html.spec.whatwg.org/multipage/sections.html#headings-and-outlines). 

Is the font too big with `###`? If so, we should fix it in the CSS file.

Author: Agree -- I'll correct the markdown.

Btw we can also force some header levels to Title Case so that happens automatically as well

- Windows 7 or later on 64bit processors
- macOS 10.12 or later on Intel processors
- macOS 11.0 or later on ARM processors (Apple Silicon)
- Ubuntu Linux 20.04 or later (other Linux distributions are supported by third parties)

> ##### Supported controllers:

Reviewer 2: ```suggestion
##### Supported Controllers
```

- Mappings for 129 MIDI controllers included
- Mappings for 16 HID controllers included

See the [full list](https://manual.mixxx.org/2.4/hardware/manuals#controllers) of supported controllers in the Mixxx Manual.

> ## Media information

Reviewer 2: "Media Information" is quite spongy since "media" can mean many things. How about "Press Information"?

```suggestion
## Press Information
```

Reviewer 3: The term "Press Information" has another meaning, which does not fit here. A "Press Information" is about specific topic to announce. The whole article could be called a "Press Information", but this is the heading of the chapter with general information. For this broader scope "Media information" is the correct wording.

Reviewer 2: This release announcement is not specifically targeted towards the press, it's also for Mixxx users, package maintainers and other interested parties.

This section is specifically targeted to DJ news channels, blogs, etc. and provides download links, image packs, logo and a press contact. Maybe you are confusing it with "news release"/press handout" (🇩🇪 "Pressemitteilung")?

My impression is that "media" instead of "press" was used here because "press" traditionally referred to printed magazines and newspapers and nowadays we also have blogs and social media. But today it is common to use "press" to mean those channels too and helps these people to spot the info easier. Hence my suggestion.

As mixxx is a "media player", the current headline is not that descriptive and without looking at the actual content the reader does not know whether it will give information tailored to journalists or if it lists "media information" like supported audio formats.

Author: as a native speaker, it's not a big deal to use one or the other.  "media" is not super confusing in the context, neither is "press"

Mixxx 2.4 is available right now:
[https://mixxx.org/download/](https://mixxx.org/download/)

The manual for Mixxx 2.4 can be found here:
[https://manual.mixxx.org/2.4/en/](https://manual.mixxx.org/2.4/en/)

> ##### Images for media use

Reviewer 2: ```suggestion
##### Images for Media Use
```

High-Resolution images of the Mixxx logo can be downloaded here:
[https://mixxx.org/press/logo](https://mixxx.org/press/logo)

Full-Resolution images of the Mixxx 2.4 screenshots above can be downloaded here:
[https://mixxx.org/screenshots](https://mixxx.org/screenshots)

> ##### Press contact

Reviewer 2: ```suggestion
##### Press Contact
```

#### Waveform Rendering Reworked

The waveform display has been rewritten, solving some longstanding performance issues, especially on macOS.

#### Apple Silicon Support

Mixxx now supports macOS ARM systems ("Apple Silicon") such as M1 natively and provides installation packages for those architectures.
For macOS users, this increases the stability and performance of Mixxx on Apple silicon.

#### Controller Backend

Modernized Javascript ES7 based controller mapping system.
HID backend rework.

### Mixxx 2.4 Factsheet

#### Supported Platforms

- Windows 7 or later on 64bit processors
- macOS 10.12 or later on Intel processors
- macOS 11.0 or later on ARM processors (Apple Silicon)
- Ubuntu Linux 20.04 or later (other Linux distributions are supported by third parties)

#### Supported Controllers

- Mappings for 129 MIDI controllers included
- Mappings for 16 HID controllers included

See the [full list](https://manual.mixxx.org/2.4/hardware/manuals#controllers) of supported controllers in the Mixxx Manual.

### Press Information

Mixxx 2.4 is available right now:
[https://mixxx.org/download/](https://mixxx.org/download/)

The manual for Mixxx 2.4 can be found here:

> [https://manual.mixxx.org/2.4/en/](https://manual.mixxx.org/2.4/en/)

Reviewer 4: To not force English 
```suggestion
[https://manual.mixxx.org/2.4](https://manual.mixxx.org/2.4)
```

#### Images For Media Use

High-Resolution images of the Mixxx logo can be downloaded here:

> [https://mixxx.org/press/logo](https://mixxx.org/press/logo)

Reviewer 4: This link is dead

Author: removed

Full-Resolution images of the Mixxx 2.4 screenshots above can be downloaded here:

> [https://mixxx.org/screenshots](https://mixxx.org/screenshots)

Reviewer 4: We need to merge the screenshot PR first

Author: order does not matter because these are all going to the 2.4 branch, which will be merged as one piece
